Core/Src/main.c

```diff
@@ -134,16 +142,45 @@ void SystemClock_Config(void)
   {
     Error_Handler();
   }
-  PeriphClkInit.PeriphClockSelection = RCC_PERIPHCLK_USART2;
-  PeriphClkInit.Usart2ClockSelection = RCC_USART2CLKSOURCE_PCLK1;
-  if (HAL_RCCEx_PeriphCLKConfig(&PeriphClkInit) != HAL_OK)
-  {
-    Error_Handler();
-  }
 }
 
 /* USER CODE BEGIN 4 */
+uint16_t readADC (uint8_t channel){
+
+
+	uint8_t receivedData[2] = {0};
```

Comment:
I think the receivedData array size should be 3. You also have used the data at index 2, so if the array is declared with size 2, that won't make sense.